lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -1213,15 +1213,23 @@ struct FpToFpOpConversion
     auto dstElementType = getElementType(op.getResult());
     bool isSrcFP32 = srcElementType.isF32();
     bool isDstFP32 = dstElementType.isF32();
-    auto cvtFunc = getConversionFunc(isSrcFP32 ? f16_ty : srcElementType,
-                                     isDstFP32 ? f16_ty : dstElementType);
     SmallVector<Value> inVals = {operands[0][0], operands[1][0], operands[2][0],
                                  operands[3][0]};
     if (isSrcFP32)
       for (Value &v : inVals)
         v = convertFp32ToFp16(loc, rewriter, v);
-    SmallVector<Value> outVals =
-        cvtFunc(loc, rewriter, inVals[0], inVals[1], inVals[2], inVals[3]);
+
+    SmallVector<Value> outVals;
+    if (srcElementType.isF16() && dstElementType.isF32()) {
+      outVals.push_back(convertFp16ToFp32(loc, rewriter, inVals[0]));
+      outVals.push_back(convertFp16ToFp32(loc, rewriter, inVals[1]));
+      outVals.push_back(convertFp16ToFp32(loc, rewriter, inVals[2]));
+      outVals.push_back(convertFp16ToFp32(loc, rewriter, inVals[3]));
+      return outVals;
+    }
+    auto cvtFunc = getConversionFunc(isSrcFP32 ? f16_ty : srcElementType,
+                                     isDstFP32 ? f16_ty : dstElementType);
+    outVals = cvtFunc(loc, rewriter, inVals[0], inVals[1], inVals[2], inVals[3]);
```

Comment:
@zoranjovanovic-ns could you wrap this change in `#ifdef USE_ROCM`?
in case this change is merged in upstream we can see that and fix it during IFU